js/script.js

```diff
@@ -89,28 +82,56 @@ function scrollToId(id) {
     }
 }
 
+function floatingButton() {
+    var lastScrollTop = 0;
+    $(window).scroll(function() {
+        var scrolling = $(window).scrollTop() + $(window).height();
+        var st = $(this).scrollTop();
+        var scrollingDown = true;
+        var $wantedSection = $('.wanted');
+        var $huntingSection = $('.hunting');
+        var wantedBottom = $wantedSection.position().top + $wantedSection.outerHeight(true);
+        var huntingBottom =  $huntingSection.position().top + $huntingSection.outerHeight(true);
+
+        scrollingDown = st > lastScrollTop ? true : false;
+        lastScrollTop = st;
+        buttonVisibility('.wanted', $wantedSection, wantedBottom, scrollingDown, scrolling);
+        buttonVisibility('.hunting', $huntingSection, huntingBottom, scrollingDown, scrolling);
+   });
+}
+
+function buttonVisibility(c, section, bottom, scrollingDown, scrolling) {
```

Comment:
Kaikissa kolmessa funktiossa parametri 'c' tarvitsee kuvaavan nimen. Nyt katsomalla funktion parametrilistaa ei voi mitenkään päätellä mikä 'c' on. Myös muut parametrit voisivat olla vähän kuvaavampia nimiltään, scrollingDown ja scrolling eivät oikein kerro mitään tyypistä tai käyttötarkoituksesta. Boolean-tyyppiset muuttujat on hyvä aloittaa is- tai has-sanalla.

Lisäksi 'c' parametri voisi suoraan olla jquery-elementti, nyt funktion käytettävyyttä huonontaa se että funktio lisää selektoriin aina luokan 'scroll__button'. Näin koodissa voisi kysyä vain element.is(':hidden').